h2/src/main/org/h2/command/Parser.java

```diff
@@ -9106,9 +9116,9 @@ private void parseCreateTableMySQLTableOptions(CreateTable command) {
 
     private void readMySQLCharset() {
         readIf(EQUAL);
-        if (!readIf("UTF8")) {
-            read("UTF8MB4");
-        }
+        readIf("UTF8");
+        readIf("UTF8MB4");
+        readIf("UTF8MB4_BIN");
```

Comment:
```Java
if (!readIf("UTF8") && !readIf("UTF8MB4")) {
    read("UTF8MB4_BIN");
}
```
or something like it.

Reply:
Would you be interested in a function in the kind of
`readAny(List<String> tokens)` or `ReadOneOf(List<String> tokens)`?

If not, I'll apply your changes ASAP :)

Reply:
No, they will be slower and they aren't normally needed.

In this specific case you actually can use `readUniqueIdentifier()`. We aren't going to do something with these encodings and collations and can accept and ignore any name. I don't see any reason to limit it to these three values.

Reply:
Even better.